src/internal/packager/helm/chart.go

```diff
@@ -172,44 +173,61 @@ func (h *Helm) GenerateChart(manifest types.ZarfManifest) (types.ConnectStrings,
 	spinner := message.NewProgressSpinner("Starting helm chart generation %s", manifest.Name)
 	defer spinner.Stop()
 
-	// Generate a new chart
+	// Generate a new chart.
 	tmpChart := new(chart.Chart)
 	tmpChart.Metadata = new(chart.Metadata)
 
-	// Generate a hashed chart name
+	// Generate a hashed chart name.
 	rawChartName := fmt.Sprintf("raw-%s-%s-%s", h.Cfg.Pkg.Metadata.Name, h.Component.Name, manifest.Name)
 	hasher := sha1.New()
 	hasher.Write([]byte(rawChartName))
 	tmpChart.Metadata.Name = rawChartName
 	sha1ReleaseName := hex.EncodeToString(hasher.Sum(nil))
 
-	// This is fun, increment forward in a semver-way using epoch so helm doesn't cry
+	// This is fun, increment forward in a semver-way using epoch so helm doesn't cry.
 	tmpChart.Metadata.Version = fmt.Sprintf("0.1.%d", config.GetStartTime())
 	tmpChart.Metadata.APIVersion = chart.APIVersionV1
 
-	// Add the manifest files so helm does its thing
+	var (
+		backtick        = []byte("`")
+		backtickReplace = []byte("`}}`{{`")
+		match           = []byte("{{")
+		prefix          = []byte("{{`")
+		suffix          = []byte("`}}")
+	)
+
+	// Add the manifest files so helm does its thing.
 	for _, file := range manifest.Files {
 		spinner.Updatef("Processing %s", file)
 		manifest := fmt.Sprintf("%s/%s", h.BasePath, file)
 		data, err := os.ReadFile(manifest)
 		if err != nil {
 			return nil, "", fmt.Errorf("unable to read manifest file %s: %w", manifest, err)
 		}
+
+		// If the template contains the match "{{", wrap it in {{` and `}} to prevent helm from trying to parse it.
+		if bytes.Contains(data, match) {
+			// Replace backticks before wrapping in {{` and `}}.
+			data = bytes.ReplaceAll(data, backtick, backtickReplace)
+			data = append(prefix, data...)
+			data = append(data, suffix...)
+		}
+
```

Comment:
I personally feel that this would be much easier if this was changed to use something like strconv.Quote https://cs.opensource.google/go/go/+/go1.20:src/strconv/quote.go;l=128 - This would handle nearly all use cases for us without needing to manually do weird things.

```
var(
		prefix          = []byte("{{")
		suffix          = []byte("}}")
)
...
data = strconv.Quote(data)
data = append(prefix, data...)
data = append(data, suffix...)
```

Reply:
You can't escape backticks within a backtick so this wouldn't work.

Reply:
Oh I see you changed the prefix/suffix to not use backtick, I can try that.

Reply:
Confirmed that escaping is not sufficient to even deploy the chart, much less pass the test.

![Screenshot 2023-02-12 at 6 33 44 PM](https://user-images.githubusercontent.com/882485/218347083-1e434f67-2920-47a2-8eed-413359957c8c.png)